File: Lumina-frontend/src/types/appConfig.ts
export interface AppConfig {
    id: number;
    key: string;
    value: string;
    description?: string;
}

export interface AppConfigCreate {
    key: string;
    value: string;
    description?: string;
}

export interface AppConfigUpdate {
    value?: string;
    description?: string;
}

File: Lumina-backend/luminalib/models/app_config.py
"""Application-wide key-value configuration model."""

from __future__ import annotations

from sqlalchemy import Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from luminalib.models.base_audit_model import AuditBase


class AppConfig(AuditBase):
    """Stores key-value configurations dynamically, replacing .env."""

    __tablename__ = "app_configs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    key: Mapped[str] = mapped_column(String(255), unique=True, index=True, nullable=False)
    value: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str | None] = mapped_column(String(255), nullable=True)
